src/main.cpp

```diff
@@ -0,0 +1,74 @@
+#include <QCoreApplication>
+#include <QDebug>
+#include <QStack>
+
+#include <QListIterator>
+#include <algorithm>
+
+int main(int argc, char *argv[])
+{
+    QCoreApplication a(argc, argv);
+
+    QList<QString> names;
+    names << "Mary" << "Ann" << "Charlynn" << "Marylynn" << "Margaret"
+          << "Kate" << "Rose" << "Gwendolyn";
+
+
+    // TODO: using Java-style iterators print the longest name from the list
+    // use QString::arg to output message with this name
+    QListIterator<QString> it(names);
+    QString longestName = it.next();
+    while (it.hasNext()) {
+        QString tmp = it.next();
+        if (tmp.size() > longestName.size()) longestName = tmp;
+    }
+    qDebug() << QString("Longest name from the list: %1").arg(longestName);
+
+    // TODO: using STL-style iterators print the shortest name from the list
+    // use QString::prepend and append to output message with this name
+    QString shortestName = names.first();
+    for (QList<QString>::const_iterator it = names.cbegin(); it < names.cend(); ++it) {
```

Comment:
Please never use "<" operator when iterating over containers.
This operation is supported only for random access iterators.